src/fp/single_word.rs

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Finite field arithmetic for any field GF(p) for which `p < 2^W`,
+//! where `W` is a specified word size.
+
+/// impl_field_single_word implements field operations for prime modulus
+/// that fits in one word of `W` bits.
+macro_rules! impl_field_single_word {
+    (
+        /// $struct_name is the name of the structure containing field
+        /// parameters and implementing field operations.
+        $struct_name:ident,
+        /// $W is the datatype of `W` bits, such that `p < 2^W`, used for
+        /// arithmetic operations.
+        $W:ty,
+        /// $W2 is the datatype of `2*W` bits used for double-length
+        /// integer operations. It must hold that `W2::BITS == 2*W::BITS`.
+        $W2:ty
```

Comment:
nit: I'm fairly certain these are not docucomments, but I could be wrong
```suggestion
// impl_field_single_word implements field operations for prime modulus
// that fits in one word of `W` bits.
macro_rules! impl_field_single_word {
    (
        // $struct_name is the name of the structure containing field
        // parameters and implementing field operations.
        $struct_name:ident,
        // $W is the datatype of `W` bits, such that `p < 2^W`, used for
        // arithmetic operations.
        $W:ty,
        // $W2 is the datatype of `2*W` bits used for double-length
        // integer operations. It must hold that `W2::BITS == 2*W::BITS`.
        $W2:ty
```